io/dagreader.go

```diff
@@ -36,6 +36,27 @@ type ReadSeekCloser interface {
 	io.WriterTo
 }
 
+// IsDir checks whether the node represents a direction
+func IsDir(n ipld.Node) bool {
+	switch n := n.(type) {
+	case *mdag.ProtoNode:
+		fsNode, err := ft.FSNodeFromBytes(n.Data())
```

Comment:
The idea would be (and I didn't indicate this correctly in the original issue, sorry) to add this function to the `FSNode` structure in `unixfs.go`, that would perform this check, so you wouldn't need to worry about the `ProtoNode` wrapper, you would just do the `switch fsNode.Type()` check.